src/main/scala/com/timeout/healthchecks/HealthChecks.scala

```diff
@@ -0,0 +1,20 @@
+package com.timeout.healthchecks
+
+import java.net.{Socket, URL}
+
+import cats.syntax.all._
+
+import scala.util.Try
+
+object HealthChecks {
+
+  def ping(host: String, port: Int) = healthCheck(s"Ping $host:$port", () => {
+    Try {
```

Comment:
ValidatedNel is just a simple type alias, but I managed to use Validated.catchNonFatal and wrap result in NonEmptyList.